libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp

```diff
@@ -30,6 +30,74 @@ bool AP_BattMonitor_SMBus::get_cycle_count(uint16_t &cycles) const
     return true;
 }
 
+// return true if serial number can be provided and fills in cycles argument
+bool AP_BattMonitor_SMBus::get_serial_number(char *serial_number, uint8_t &serial_name_len) const {
+
+    if (_serial_number != -1) {
+    	snprintf(serial_number, serial_name_len, "%i", (signed) _serial_number);
```

Comment:
Signed?  Really?
```suggestion
    	snprintf(serial_number, serial_name_len, "%u", (unsigned) _serial_number);
```

Reply:
_serial_number is an int32_t : What is the best way to deal with the cast? I'm rusty on such things.